cern_access/indico_cern_access/util.py

```diff
@@ -269,7 +269,15 @@ def withdraw_event_access_request(req):
 def get_random_reservation_code():
     """Generate random reservation code for data required by ADaMS API."""
     charset = 'ABCDEFGHIJKLMNPQRSTUVWXYZ123456789'
-    return 'I' + ''.join(random.sample(charset, 6))
+    reservation_code = ''
+
+    while (
```

Comment:
```
while True:
    code = ...
    if <not unique>:
        break
```

I think that's nicer. It also most likely lets you avoid wrapping the if criterion

Reply:
Good suggestion, and actually instead of breaking I'll just return the reservation_code right away